### ISSUE TITLE
Incorrect timing for read key state transitions

### ISSUE THREAD

**Issue by tomato42** [tomato42/tlsfuzzer#405]
# Bug Report

## System Information

| Type                     | Version/Name |
| ------------------------ | ------------ |
| Operating System Name    | Fedora  |
| Operating System Version |  26  |
| Python version           | 2.7.15  |
| tlslite-ng version       | 0.8.0-alpha11  |
| tlsfuzzer version      | d6144454967b |

## Problem description

The changes to read and write keys in handshake are happening at wrong places

## Describe how to reproduce the problem

no reproducer, just state transitions

## Expected behaviour

according to the draft:
```
   Any records following a Finished message MUST be encrypted under the
   appropriate application traffic key as described in Section 7.2.  In
   particular, this includes any alerts sent by the server in response
   to client Certificate and CertificateVerify messages.
```
In other words, the read side need to transfer to application layer keys as soon as the server Finished is received, currently it does so in FinishedGenerator  object

## Include errors or backtraces
no backtrace

## Additional context

[TLS1.3 draft 28, section 2](https://tools.ietf.org/html/draft-ietf-tls-tls13-28#page-21)
